### PR TITLE
Enable arithmetic with Sage Integers & Rationals

### LINKED ISSUE #9: arithmetic does not play well with Sage integers
Currently
```
sage: from pyeantic import eantic
sage: K = eantic.renf_class('x^2 - 2', 'x', '[1.4142 +/- 0.0001]')
sage: a = K.gen()
sage: a + 1
Traceback (most recent call last):
<ipython-input-4-6bb917f1f9ff> in <module>()
----> 1 a + Integer(1)

/opt/sage/sage-py3-gcc/local/lib/python3.7/site-packages/pyeantic/cppyy_eantic.py in binary(lhs, rhs, op)
     58                 return type(x).__cppname__ if hasattr(type(x), '__cppname__') else type(x).__name__
     59             def binary(lhs, rhs, op = op):
---> 60                 return cppyy.gbl.eantic.boost_binary[cppname(lhs), cppname(rhs), op](lhs, rhs)
     61             def inplace(lhs, *args, **kwargs): raise NotImplementedError("inplace operators are not supported yet")
     62             setattr(proxy, "__%s__"%n, binary)

TypeError: cannot resolve method template call for 'boost_binary'
```
That is to be expected but is not very user friendly. For cypari2/gmpy2 that also interacts with Sage objects there are conversion operators `__pari__`, `__mpz__`, `__mpq__`, etc implemented directly on Sage objects.

### PR DESCRIPTION
**Pull request by saraedum**
fixes #9

### CODE REVIEW

**Comment by videlec** [comment]
If you are going through strings, why are you trying the `hasattr(x, '__mpq__')`? A faster and simpler test for the conversion could be
```
if v.__mod__.startswith('sage.'):
    # assumes v is a SageMath object and do a lazy conversion via string
    v = str(v)
```

**Comment by saraedum** [comment]
> If you are going through strings, why are you trying the `hasattr(x, '__mpq__')`? A faster and simpler test for the conversion could be
> 
> ```
> if v.__mod__.startswith('sage.'):
>     # assumes v is a SageMath object and do a lazy conversion via string
>     v = str(v)
> ```

How would I know that `str` is understood by renf_elem_class in the right way? That's why I go through mpq. For the moment, I only want to support rationals (or anything that can be turned into one exactly.)

File: src/pyeantic/pyeantic/cppyy_eantic.py [hunk 1]
```diff
@@ -139,11 +139,29 @@ def py_make_renf_elem_class(*args):
         if isinstance(v, eantic.renf_class):
             return eantic.renf_elem_class(v)
         else:
+            v = sage_to_gmp(v)
             return eantic.make_renf_elem_class(v)
     elif len(args) == 2:
         K, v = args
         if isinstance(v, (tuple, list)):
             v = cppyy.gbl.std.vector[int](v)
```
Comment (videlec) [inline src/pyeantic/pyeantic/cppyy_eantic.py]: What about a vector of Sage integers/rationals?

Comment (saraedum) [reply src/pyeantic/pyeantic/cppyy_eantic.py]: True.

Comment (saraedum) [reply src/pyeantic/pyeantic/cppyy_eantic.py]: When you say "vector" you mean "Python list"?

Comment (videlec) [reply src/pyeantic/pyeantic/cppyy_eantic.py]: Yes. For now I am not sure there is a need to support something more general than tuple/list.

**Comment by videlec** [comment]
Also, I noted that in `sage_conversion.sage_nf_elem_to_eantic` I made the function converts much more than what it is intended to. I guess that we should make it less flexible (at the same time we make `py_make_renf_elem_class` more flexible).

**Comment by saraedum** [comment]
> Also, I noted that in sage_conversion.sage_nf_elem_to_eantic I made the function converts much more than what it is intended to. I guess that we should make it less flexible (at the same time we make py_make_renf_elem_class more flexible).

Ok. Will do.

**Pull request merged**